templates/multistage-buildconfig.yaml

```diff
@@ -0,0 +1,99 @@
+---
+apiVersion: template.openshift.io/v1
+kind: Template
+metadata:
+  annotations:
+    description: Template to produce an imagestream and buildconfig for a Jlinked image
+  name: jlinked-image-template
+objects:
+- apiVersion: image.openshift.io/v1
+  kind: ImageStream
+  metadata:
+    name: jlinked-image
+  spec:
+    lookupPolicy:
+      local: false
+- apiVersion: build.openshift.io/v1
+  kind: BuildConfig
+  metadata:
+    name: ubi9-jlinked-image-buildconfig
+  spec:
+    source:
+      dockerfile: |
+        #Stage-1: ubi9-jlinked-image is builder image + application + jlinked JVM
+        # install additional system dependencies (for ubi-micro) to /mnt/jrootfs
+        FROM -
+        USER 0
+        RUN mkdir -p /mnt/jrootfs
+        RUN microdnf install --installroot /mnt/jrootfs --releasever 9 --setopt install_weak_deps=0 --nodocs -y \
+            --config=/etc/dnf/dnf.conf \
+            --noplugins \
+            --setopt=cachedir=/var/cache \
+            --setopt=reposdir=/etc/yum.repos.d \
+            --setopt=varsdir=/etc/dnf/vars \
+            grep gawk
+        RUN rm -rf /mnt/jrootfs/var/cache/* /mnt/jrootfs/var/lib/rpm /mnt/jrootfs/var/lib/dnf
+      strategy:
+        dockerStrategy:
+          from:
+            kind: ImageStreamTag
+            name: quay.io/repository/jdowland/jlink:ubi9-jlinked-image
+      output:
+        to:
+          kind: ImageStreamTag
+          name: ubi9-jlinked-image:latest
+      triggers:
+        - type: ConfigChange
+        - type: ImageChange
+          imageChange:
+            from:
+              kind: ImageStreamTag
+              name: quay.io/repository/jdowland/jlink:ubi9-jlinked-image  # ImageStreamTag to trigger the build (stage-1 in Dockerfile)
+- apiVersion: build.openshift.io/v1
+  kind: BuildConfig
+  metadata:
+    name: lean-runtime-buildconfig
+  spec:
+    source:
+      dockerfile: |
+        #Stage-2:copy application JAR and jlinked JRE to runtime image
+        FROM -
```

Comment:
this doesn't need to be parameterized I don't think. use `FROM registry.access.redhat.com/ubi9/ubi-micro`. (I wonder if we should actually define an ImageStream for ubi-micro or not.)